### PR TITLE
refactor

### PR DESCRIPTION
**Pull request by Irtiza90**
- variable name changes, and more slots
- split the function `run` into `run` and `run_debug`
- make `types.Context` and `utils.Pos` dataclasses

### CODE REVIEW

File: Cyan/exceptions.py [hunk 1]
```diff
@@ -2,68 +2,70 @@
...
         self.ecode = ""
 
     def set_ecode(self, ecode):
         self.ecode = ecode
         return self
 
     def set_pos(self, pos_start, pos_end):
-        self.pos_start = pos_start
-        self.pos_end = pos_end
+        self.start_pos = pos_start
```
Comment (was07) [inline Cyan/exceptions.py:19]: Yeah I guess that makes more sense

File: Cyan/types.py [hunk 2]
```diff
@@ -350,7 +355,7 @@ def is_truthy(self) -> Bool:
         return Bool(self.value)
 
     # converters
-    def to_Number(self) -> RTResult:
+    def to_number(self) -> RTResult:
```
Comment (was07) [inline Cyan/types.py]: That isn't going to work. the to_{type} method names has to be exactly same as the real type name, it has to be to_Number and not to_number

**Review by was07 — CHANGES_REQUESTED** [review]

File: Cyan/types.py (hunk 2 above)
Comment (Irtiza90) [reply Cyan/types.py]: there's multiple of these `to_type` methods do they also need to be changed?
or is it only this one

Comment (was07) [reply Cyan/types.py]: They all need to be the exact as the type name, but I don't see any other `to_type` method renamed

**Review by was07 — APPROVED** [review]
Nice work

**Pull request merged**